velox/functions/prestosql/aggregates/MinMaxAggregates.cpp

```diff
@@ -380,11 +413,9 @@ class NonNumericMinMaxAggregateBase : public exec::Aggregate {
       if (decoded.isNullAt(i)) {
         return;
       }
+
       auto accumulator = value<SingleValueAccumulator>(groups[i]);
-      if (!accumulator->hasValue() ||
-          compareTest(accumulator->compare(decoded, i))) {
-        accumulator->write(baseVector, indices[i], allocator_);
-      }
+      updateAccumulator(accumulator, decoded, i, compareTest);
```

Comment:
I'm wonder if it is better to keep the original code, but replace `accumulator->compare(decoded, i)` 

```
if (!accumulator->hasValue() ||
          compareTest(compare(accumulator, decoded, i))) {
...


bool compare(accumulator, decoded, index) {
static const CompareFlags kCompareFlags{
        true, // nullsFirst
        true, // ascending
        false, // equalsOnly
        CompareFlags::NullHandlingMode::StopAtNull};

    auto result = accumulator->compare(decoded, i, kCompareFlags);
    VELOX_USER_CHECK(
        result.has_value(),
       ...);

    return result.value();
}
```